test/Driver/batch_mode_with_WMO_or_index.swift

```diff
@@ -13,6 +13,12 @@
 // RUN: %FileCheck -check-prefix CHECK-INDEX %s <%t/stderr_batch_index
 // CHECK-INDEX: warning: ignoring '-enable-batch-mode' because '-index-file' was also specified
 //
+// RUN: %swiftc_driver -disable-batch-mode -index-file  %S/../Inputs/empty.swift -### 2>%t/stderr_nobatch_index | %FileCheck %s
```

Comment:
How about adding a comment like the one on line 22, briefly describing the reason for the test? As a newbie, I've found such to help a lot.

Reply:
Sure, I'll add a follow-up commit explaining what my goal is here.  Roughly:

The following test is verifying that `-disable-batch-mode` overrides an earlier `-enable-batch-mode` and silences the warning about mixing batch mode with `-index-file`.  Tools that take an existing command line and add `-index-file` can add `-disable-batch-mode` without having to otherwise interpret the arguments.